src/utils/daos_hdlr.c

```diff
@@ -3142,6 +3108,9 @@ dm_parse_path(struct file_dfs *file, char *path, size_t path_len,
 		} else {
 			strncpy(path, dattr.da_rel_path, path_len);
 		}
+	} else if (rc == ENOMEM) {
+		/* TODO: Take this path of rc != ENOENT? */
+		return rc;
```

Comment:
Interesting... I don't think we can just use != ENOENT because the `else if` below still needs to be checked. Basically, the path passed to duns_resolve_path might be a UNS path, DAOS path, or a regular POSIX path. The reason for the `else if` below is really for an early exit, instead of assuming an invalid `daos://` path is POSIX and then getting more errors later.

Reply:
This code does need an update, there are many reasons duns_resolve_path() should fail and almost none of them are handled correctly.

Reply:
Makes sense. This comment at least reminds us that it needs work

Reply:
@ashleypittman can you be more specific about which failures are missing here in this function and how they should be handled? Or point me to some DAOS code that responds to failures of duns_resolve_path the way you would like to see it?

Reply:
Looking at the code if there's any error from duns_resolve_path() then it'll do the strcmp and return EINVAL if that doesn't match, which means that many classes of error will be failed with EINVAL.  I think it should probably only do the stncmp() at all if rc == ENOENT but you'd have to check duns_resolve_path() to make sure and check any tests that we have.  I'd like to avoid getting too far into the weeds in this PR.

Reply:
@ashleypittman yeah, that's fine. I can look into it.